tests/quarkus/src/test/java/io/hawt/tests/quarkus/suites/CucumberQuarkusTest.java

```diff
@@ -0,0 +1,239 @@
+package io.hawt.tests.quarkus.suites;
+
+import org.junit.jupiter.api.DynamicContainer;
+import org.junit.jupiter.api.DynamicNode;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.platform.console.ConsoleLauncher;
+
+import org.opentest4j.TestSkippedException;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import java.net.URI;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.cucumber.core.backend.ObjectFactory;
+import io.cucumber.core.eventbus.EventBus;
+import io.cucumber.core.feature.FeatureParser;
+import io.cucumber.core.filter.Filters;
+import io.cucumber.core.gherkin.Pickle;
+import io.cucumber.core.options.CommandlineOptionsParser;
+import io.cucumber.core.options.Constants;
+import io.cucumber.core.options.CucumberOptionsAnnotationParser;
+import io.cucumber.core.options.CucumberProperties;
+import io.cucumber.core.options.CucumberPropertiesParser;
+import io.cucumber.core.options.RuntimeOptions;
+import io.cucumber.core.options.RuntimeOptionsBuilder;
+import io.cucumber.core.plugin.Options;
+import io.cucumber.core.plugin.PluginFactory;
+import io.cucumber.core.plugin.Plugins;
+import io.cucumber.core.plugin.PrettyFormatter;
+import io.cucumber.core.runner.Runner;
+import io.cucumber.core.runtime.CucumberExecutionContext;
+import io.cucumber.core.runtime.ExitStatus;
+import io.cucumber.core.runtime.FeaturePathFeatureSupplier;
+import io.cucumber.core.runtime.FeatureSupplier;
+import io.cucumber.core.runtime.ObjectFactorySupplier;
+import io.cucumber.core.runtime.TestCaseResultObserver;
+import io.cucumber.core.runtime.TimeServiceEventBus;
+import io.cucumber.java.JavaBackendProviderService;
+import io.cucumber.plugin.event.EventHandler;
+import io.cucumber.plugin.event.Status;
+import io.cucumber.plugin.event.TestStepFinished;
+import io.quarkiverse.cucumber.QuarkusCucumberOptionsProvider;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public abstract class CucumberQuarkusTest {
```

Comment:
Copy from quarkus-cucumber to allow for necessary changes for adapting to Quarkus 2.x